File: ruby/testcode/weakhash_038.rb
require_relative 'shared'

begin
  require 'bcrypt'
rescue LoadError
  # bcrypt gem not available -- syntax-checked only
end

class User
  include BCrypt

  def self.create_with_password(pw)
    BCrypt::Password.create(pw)
  end
end

# vuln-code-snippet start ruby_weakhash_has_secure_password
def register_user(req)
  password = req.param('password')
  hash = User.create_with_password(password)  # vuln-code-snippet safe-line ruby_weakhash_has_secure_password
  BenchmarkResponse.json({ ok: true, hash: hash })
end
# vuln-code-snippet end ruby_weakhash_has_secure_password

File: php/testcode/weakhash_038.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakhash_hmac_sha256_mac
function weakhash038(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->bodyStr();
    $key = getenv('MAC_SECRET');
    $mac = hash_hmac('sha256', $data, $key); // vuln-code-snippet safe-line php_weakhash_hmac_sha256_mac
    return BenchmarkResponse::ok($mac);
}
// vuln-code-snippet end php_weakhash_hmac_sha256_mac
